Common++/src/Logger.cpp

```diff
@@ -4,55 +4,50 @@
 namespace pcpp
 {
 
-Logger::Logger() : m_LogsEnabled(true), m_LogPrinter(&defaultLogPrinter)
-{
-	m_LastError.reserve(200);
-	for (int i = 0; i<NumOfLogModules; i++)
-		m_LogModulesArray[i] = Info;
-}
-
-std::string Logger::logLevelAsString(LogLevel logLevel)
-{
-	switch (logLevel)
+	Logger::Logger() : m_LogsEnabled(true), m_LogPrinter(&defaultLogPrinter)
 	{
-	case Logger::Error:
-		return "ERROR";
-	case Logger::Info:
-		return "INFO";
-	default:
-		return "DEBUG";
+		m_LastError.reserve(200);
+		for (int i = 0; i < NumOfLogModules; i++)
+			m_LogModulesArray[i] = Info;
 	}
-}
 
-void Logger::defaultLogPrinter(LogLevel logLevel, const std::string& logMessage, const std::string& file, const std::string& method, const int line)
-{
-	std::ostringstream sstream;
-	sstream << file << ": " << method << ":" << line;
-	std::cout << std::left
-		<< "["
-		<< std::setw(5) << Logger::logLevelAsString(logLevel) << ": "
-		<< std::setw(45) << sstream.str()
-		<< "] "
-		<< logMessage << std::endl;
-}
-
-std::ostringstream * Logger::internalCreateLogStream()
-{
-	return new std::ostringstream ();
-}
+	std::string Logger::logLevelAsString(LogLevel logLevel)
+	{
+		switch (logLevel)
+		{
+		case Logger::Error:
+			return "ERROR";
+		case Logger::Info:
+			return "INFO";
+		default:
+			return "DEBUG";
+		}
+	}
 
-void Logger::internalPrintLogMessage(std::ostringstream* logStream, Logger::LogLevel logLevel, const char* file, const char* method, int line)
-{
-	std::string logMessage = logStream->str();
-	delete logStream;
-	if (logLevel == Logger::Error)
+	void Logger::defaultLogPrinter(LogLevel logLevel, const std::string &logMessage, const std::string &file,
+								   const std::string &method, const int line)
 	{
-		m_LastError = logMessage;
+		std::ostringstream sstream;
+		sstream << file << ": " << method << ":" << line;
+		std::cout << std::left << "[" << std::setw(5) << Logger::logLevelAsString(logLevel) << ": " << std::setw(45)
+				  << sstream.str() << "] " << logMessage << std::endl;
```

Comment:
I actually like the previous version better, but not sure if there is a way to keep it...

Reply:
I cannot find the solution for this.